File: app/src/main/java/com/past/music/utils/DensityUtil.java
package com.past.music.utils;

import android.content.Context;

/**
 * Created by gaojin on 2017/3/25.
 */

public class DensityUtil {
    /**
     * 将sp值转换为px值，保证文字大小不变
     *
     * @param spValue
     * @return
     */
    public static int sp2px(Context context, float spValue) {
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (spValue * fontScale + 0.5f);
    }
}

File: app/src/main/java/com/se/music/utils/DensityUtil.kt
package com.se.music.utils

import android.content.Context

/**
 * Created by gaojin on 2018/3/6.
 */
class DensityUtil {

    companion object {
        /**
         * 将sp值转换为px值，保证文字大小不变
         *
         * @param spValue
         * @return
         */
        fun sp2px(context: Context, spValue: Float): Int {
            val fontScale = context.resources.displayMetrics.scaledDensity
            return (spValue * fontScale + 0.5f).toInt()
        }

        /**
         * 根据手机的分辨率从 dp 的单位 转成为 px(像素)
         */
        fun dip2px(context: Context, dpValue: Float): Int {
            val scale = context.resources.displayMetrics.density
            return (dpValue * scale + 0.5f).toInt()
        }
    }
}